File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int  l = 0;
        int  r = 1;
        int max_profit = 0;
        int n = prices.size();

        while(r < n){
            if (prices[r] < prices[l]) {
                l = r; 
            } else {
                max_profit = max(max_profit, prices[r] - prices[l]);
            }
            r++;
        }

        return max_profit;
        
    }
};

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        l = 0
        r = 1
        maxProfit = 0
        n = len(prices)

        while r<n:
            if prices[r] < prices[l]: l=r
            else: maxProfit=max(maxProfit, prices[r] - prices[l])
            r+=1
        
        return maxProfit
        